turbinia/workers/__init__.py

```diff
@@ -609,14 +628,24 @@ def run_wrapper(self, evidence):
 
     log.debug('Task {0:s} {1:s} awaiting execution'.format(self.name, self.id))
     evidence = evidence_decode(evidence)
+    try:
+      self.result = self.setup(evidence)
+      self.result.update_task_status(self, 'Queued')
+    except TurbiniaException as exception:
+      message = (
+          '{0:s} Task setup failed with exception: [{1!s}]'.format(
+              self.name, exception))
+      # Logging explicitly here because the result is in an unknown state
+      trace = traceback.format_exc()
+      log.error(message)
+      log.error(trace)
```

Comment:
Sorry, I misled you a bit with my last comment here.  We do actually need to get a valid `TurbiniaTaskResult` object somewhere here so that we can return these errors (the other checks were still fine to remove).   

Since we can't really continue after the error, for the short term here, let's just check if self.result exists, and if not, create a new `TurbiniaTaskResult` object (similar to in `validate_result`), call `self.result.setup()` and then use `self.log(message, logging.error, traceback_=trace)` to log, and finally return `self.result.serialize()`.

Later in a second PR I'd like to refactor the places where `TurbiniaTaskResult` is generated to a single method, and make it simpler so that we don't have to run the `setup()` methods during exception handling because they do non-trivial things and run the risk of throwing more exceptions during exception handling, which means that we won't get valid results back to the server (and the task will stay pending indefinitely).

Reply:
Done, had to change testTurbiniaTaskRunWrapperSetupFail to prevent failure since validate_result() never gets called this way.

Reply:
Cool, thanks, I'll refactor the task result checking/generation code in a separate PR then.